mod-source-record-storage-server/src/main/java/org/folio/dao/PostgresClientFactory.java

```diff
@@ -169,9 +171,13 @@ private static PgPool getCachedPool(Vertx vertx, String tenantId) {
     }
 
     Integer maxPoolSize = postgresConfig.getInteger(DB_MAXPOOLSIZE, DB_MAXPOOLSIZE_DEFAULT_VALUE);
+    int connectionTimeout = Integer.parseInt(System.getenv().getOrDefault(CONNECTION_TIMEOUT, DEFAULT_CONNECTION_TIMEOUT_VALUE));
     LOG.info("getCachedPool:: Creating new database connection for tenant {} with poolSize {}", tenantId, maxPoolSize);
     PgConnectOptions connectOptions = getConnectOptions(tenantId);
-    PoolOptions poolOptions = new PoolOptions().setMaxSize(maxPoolSize);
+    PoolOptions poolOptions = new PoolOptions()
+      .setConnectionTimeout(connectionTimeout)
+      .setIdleTimeoutUnit(TimeUnit.SECONDS)
```

Comment:
I think you meant `setConnectionTimeoutUnit` here instead of `setIdleTimeoutUnit`. The default is still "seconds" for `setConnectionTimeoutUnit` so the intent is still there. But we should change to accurately depict what we mean.

Reply:
thank you for spotting this, I will correct it